### DISCUSSION TITLE
Loading a new document layout without having to redeploy

### DISCUSSION THREAD

**Discussion by dan5602** [QuestPDF/QuestPDF#598] [Q&A] [answered]
Has anyone thought about the possibility of loading a new document layout without having to redeploy the whole application?

Example: Each customer needs a slightly different document layout for an invoice. Currently, we would have to create the document layout in code and then redeploy the app for the new document layout to be available.

Does anyone have any thoughts or tips on this?

**Comment by MarcinZiabek** [comment]
Hi 😁 

Would you please share more insights about the similarities of your client's documents? 

You can often use flow control features to alter the layout conditionally, e.g., if statements and for loops in c#. They could be used to preconfigure certain layout structures.

**Comment by DawidRobertNowak** [comment, marked as answer]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out.

As @MarcinZiabek  suggested, QuestPDF's code-first approach makes this achievable using standard C# flow control — conditional statements, loops, and configuration-driven logic can be used to alter layouts per customer without redeployment. You may also find the [Components](https://www.questpdf.com/concepts/code-patterns/components.html) and [Content Styling](https://www.questpdf.com/concepts/code-patterns/content-styling.html) patterns helpful for organizing reusable layout logic.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
